src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -93,15 +94,29 @@ public static Phone parsePhone(String phone) throws ParseException {
      *
      * @throws ParseException if the given {@code dob} is invalid.
      */
-    public static DateOfBirth parseDob(String dob) throws ParseException {
+    public static DateOfBirth parseDob(String dob, boolean isNaAllowed) throws ParseException {
         requireNonNull(dob);
+        requireNonNull(isNaAllowed);
         String trimmedDob = dob.trim();
-        if (!DateOfBirth.isValidDateOfBirth(dob)) {
+        if (!DateOfBirth.isValidDateOfBirth(dob, isNaAllowed)) {
             throw new ParseException(DateOfBirth.MESSAGE_CONSTRAINTS);
         }
+        if (dob.equals(EMPTY_ARGUMENT) & isNaAllowed) {
+            return DateOfBirth.getEmptyDateOfBirth();
+        }
```

Comment:
I feel that the logic to handle the `isNaAllowed` flag is better suited to be in the `DateOfBirth.isValidDateOfBirth()` method. It looks like this is breaking abstraction since `parseDob()` has to determine what is a valid DOB instead of leaving that decision to the `isValidDateOfBirth()` method.

Qiaoran faced a similar [issue](https://github.com/AY2223S1-CS2103-F09-2/tp/pull/52#discussion_r991807613) which I commented on as well. You can take a look and see if refactoring is appropriate for this case.

Reply:
I feel that it is fine to leave it in the parser as the parser's job is to dissect the command, and in this case once it has gotten the argument for dob it will also look at the overall command keyword (if its an edit or add command, which is the origin of the isNaAllowed as it is only btwn different commands that it changes) in order to then pass it off seperately to the DOB class for it to do the constructing. However, I do see that gender has been changed to instead process the command type within the gender class (which also seems like an abstraction peculiarity since i dont believe these field classes should be caring about the command type) and that either way there will be some abstraction arguments to be made for both sides, so for the sake of keeping it consistent i will change it to be inside of the DOB class.